### DISCUSSION TITLE
Problem when STM32H7 cache enabled

### DISCUSSION THREAD

**Discussion by gawelkot** [cesanta/mongoose#2643] [Ask questions, get answers] [answered]
Hello,
I am trying to run Mongoose with built-in IP stack using Cube and FreeRTOS. I don't have any problems when cache is disabled in STM32H7 mcu. When I enable cache network is not working at all. Also I didn't find any example with cache enabled. I guess problem is with MPU configuration for RX and TX buffers which are not updated as it is in lwip IP stack. I tried to allocate below variables in non-cachable MPU regions but it's still not working:
```
static volatile uint32_t s_rxdesc[ETH_DESC_CNT][ETH_DS];  // RX descriptors
static volatile uint32_t s_txdesc[ETH_DESC_CNT][ETH_DS];  // TX descriptors
static uint8_t s_rxbuf[ETH_DESC_CNT][ETH_PKT_SIZE];       // RX ethernet buffers
static uint8_t s_txbuf[ETH_DESC_CNT][ETH_PKT_SIZE];       // TX ethernet buffers
```

Do you have any expierience with such MCU configuration? I would appreciate any help with this topic.

**Comment by scaprile** [comment]
Our experience is in our examples.
We don't enable cache as we do not pay attention to where memory has been allocated nor invalidate the cache, nor we use the MPU; as you've already seen.

H7's ETH can only access certain RAM regions, if descriptors or buffers are allocated in regions not accessible by the peripheral... that won't work.

We don't have any plans for addressing caches nor MPU. If you are a paying customer or want to be and this is important to you, please contact Cesanta.

**Comment by alexachso** [comment]
Hi, @gawelkot 
Did you by any chance got it working with the cache enabled?

**Reply by scaprile** [reply, marked as answer]
> Hi, @gawelkot Did you by any chance got it working with the cache enabled?

Please do not necropost
Please see our [documentation](https://mongoose.ws/documentation/),  and follow the guidelines in our [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples).
https://mongoose.ws/articles/stm32-ethernet-and-cache/
